server/instance_cloud_oauth.go

```diff
@@ -99,6 +99,11 @@ func (ci *cloudOAuthInstance) getClientForConnection(connection *Connection) (*j
 	oauth2Conf := ci.GetOAuthConfig()
 	ctx := context.Background()
 	tokenSource := oauth2Conf.TokenSource(ctx, connection.OAuth2Token)
+	if ci.JWTInstance != nil && tokenSource == nil {
+		ci.Plugin.API.LogDebug("Returning a JWT token client in case the stored JWT instance is not nil and the user's oauth token is nil")
```

Comment:
```suggestion
		ci.Plugin.API.LogDebug("Returning a JWT token client since the stored JWT instance is not nil and the user's oauth token is nil")
```